command/connect/envoy/bootstrap_config.go

```diff
@@ -305,6 +307,18 @@ func (c *BootstrapConfig) generateStatsSinkJSON(name string, addr string) (strin
 	}`, nil
 }
 
+func (c *BootstrapConfig) statsSinkEnvMapping(s string) string {
+	allowedStatsSinkEnvVars := map[string]bool{
```

Comment:
this seemed like the simplest place to put this allowlist. happy to move elsewhere if desired. 
the original implementation explicitly didn't support expanding arbitrary env vars, so the purpose here to minimize the potential side effects while still providing the ability to dynamically set this. happy to remove this functionality and just use a plain `os.ExpandEnv` if that's preferable.